src/main/java/org/apache/mesos/state/SchedulerState.java

```diff
@@ -0,0 +1,135 @@
+package org.apache.mesos.state;
+
+import com.google.protobuf.TextFormat;
+import org.apache.mesos.Protos.*;
+import org.apache.mesos.curator.CuratorStateStore;
+import org.apache.mesos.offer.TaskException;
+import org.apache.mesos.offer.TaskUtils;
+import org.apache.mesos.reconciliation.TaskStatusProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+
+/**
+ * Read/write interface for the state of a Scheduler.
+ */
+public class SchedulerState implements TaskStatusProvider {
+    private static final Logger log = LoggerFactory.getLogger(SchedulerState.class);
+    private static final String SUPPRESSED_KEY = "suppressed";
+
+    protected final AbstractStateStore stateStore;
+
+    public SchedulerState(String frameworkName, String mesosZkURI) {
```

Comment:
It feels odd pulling in one of the concrete state store types here. Maybe just omit this constructor in favor of the following one?